### DISCUSSION TITLE
Social Login(Google)

### DISCUSSION THREAD

**Discussion by cheadevit** [Laravel-Backpack/community-forum#625] [Q&A (Help)] [answered]
Hello Guys,

I’m trying to implement Social Login(Google) in my project, for the first time it's working after I run `composer update`, and now when I try to deploy it again I got some error as below:

`"cURL error 6: getaddrinfo() thread failed to start (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://www.googleapis.com/oauth2/v4/token" // app/Http/Controllers/GoogleController.php:58`

<img width="705" alt="Screenshot 2023-08-08 at 15 51 50" src="https://github.com/Laravel-Backpack/community-forum/assets/52439329/ded3dbdd-c974-4cf1-a14b-ea11905201e1">

Here is my project version
### PHP VERSION:
PHP 8.1.16 (cli) (built: Feb 14 2023 18:34:53) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.16, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.16, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
10.17.1.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.1.1
backpack/crud: 6.1.5
backpack/generators: v4.0.2
backpack/pro: 2.0.10
backpack/theme-tabler: 1.0.8

Please helps, appreciate it.

**Comment by promatik** [comment]
Hi @cheadevit!

This is hard to tell, most probably it's not related with backpack.
Can you show us your `GoogleController.php` line 58?

The composer update you run, updated some dependencies and one or some of them is causing this new issue. Can you show us the update log? what were the dependencies updated?

If there was a major version update on any of your packages, it's most likely that what's causing the issue.
Please be sure to follow any upgrade guide of any third party packages.

**Reply by cheadevit** [reply]
Hello @promatik 

Here is the error message on line 58 `Auth::login($newUser);`
Here is my full code in `GoogleController.php`

```
<?php

namespace App\Http\Controllers;
use Laravel\Socialite\Facades\Socialite;
use Exception;
use App\Models\User;
use Illuminate\Support\Facades\Auth;

use Illuminate\Http\Request;

class GoogleController extends Controller
{
    public function redirectToGoogle()
    {
        // $error = 'error';
        // return redirect()->route('backpack.auth.login')->with('error','Not allow ');
        return Socialite::driver('google')->redirect();
    }
        
    /**
     * Create a new controller instance.
     *
     * @return void
     */
    public function handleGoogleCallback()

    {   // Try to exception
        try {
            $user = Socialite::driver('google')->user(); // google gmail

            if (explode("@", $user->email)[1] != 'example.com'){
                return redirect()->route('backpack.auth.login')->with('error','Email Not Allow');
            }

            $finduser = User::where('email', '=', $user->email)->first();
       
            if(isset($finduser) == true){
       
                $updateUser = User::where('email', '=', $user->email)->update([
                    'email' => $user->email,
                    'google_id'=> $user->id,
                ]);

                Auth::login($finduser);
      
                return redirect()->intended('admin/dashboard');
       
            }else{
                $newUser = User::create([
                    'name' => $user->name,
                    'email' => $user->email,
                    'google_id'=> $user->id,
                    'password' => encrypt('*****')
                ]);

                $newUser->assignRole(2);

                Auth::login($newUser);
      
                return redirect()->intended('admin/dashboard');
            }
      
        } catch (Exception $e) {
            dd($e->getMessage());
        }
    }
}
```
Thanks for your response.

**Reply by promatik** [reply]
@cheadevit, I don't see any issue in the code...
Let's try to figure out something more;
1) Where is this code running? is it local or on some server? is it under a VPN? 
At some place something can be blocking the request to google apis...

2) You mentioned an upgrade, have you done it? Was any package updated? like socialite or something?
If you revert that upgrade will it work again?
Make sure to follow upgrade guides, sometimes it happens that a config file must be updated, if somehow that google api depends on some config, the issue can be there.

**Reply by cheadevit** [reply]
Hi @promatik Thank you.

My project(docker container) is running on an AWS ec2 instance.
Even try to use curl with an external API like Slack it still shows the same error.

```"cURL error 6: getaddrinfo() thread failed ```

**Reply by promatik** [reply, marked as answer]
Ok! so we're getting somewhere ...
Did you setup your ec2 instance firewall? or ufw on your droplet?

Make sure;
1) AWS firewall rules are not blocking the access of your app, or
2) EC2 droplet has UFW properly configured (perhaps disabled)

I would bet one of this are restricting your access.
I always use very strict rules on my droplets, directly on the service provider (Digital Ocean), and for this reason I usually do not use UFW.
